source/agora/node/Ledger.d

```diff
@@ -842,6 +842,34 @@ public class Ledger
             .map!(idx => this.storage.readBlock(Height(idx)));
     }
 
+    /***************************************************************************
+
+        Get a block at the provided block height.
+
+        Params:
+            height = the block height
+
+        Returns:
+            the block
+
+    ***************************************************************************/
+
+    public Block getBlockAtHeight (Height height) @safe nothrow
+    {
+        if (height == this.last_block.header.height)
+            return this.last_block;
+        else
+            try
+            {
+                return this.storage.readBlock(Height(height));
+            }
+            catch(Exception e)
+            {
+                log.fatal("Block failed to be loaded from storage for height {}: {}", height, e);
+                assert(0);
+            }
+    }
+
```

Comment:
Nope, let's not add more storage dependence to the ledger. Additionally, no `assert(0)`

Reply:
Removed as not needed anymore.